flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java

```diff
@@ -68,7 +68,6 @@ public void testSuccessfulProgramAfterFailure() {
 	
 			try {
 				runKMeans(cluster.getLeaderRPCPort());
-				fail("This program execution should have failed.");
```

Comment:
This case will fail if not remove it. 

I guess the buffer amount is not enough for this case before, because some are shared by buffer provider in handler. After remove the buffer provider from handler, the buffer is enough now. So the `runKmeans` can execute correctly, not enter catch part or fail process.

I am not familiar with this case and not sure whether my understanding is right.